### PR TITLE
search: Remove transitive Monaco dependency from CodeMirror query input

### PR DESCRIPTION
**Pull request by fkling**
**tl;dr:** Reduces bundle size in development mode, which reduces load times when running integration tests locally (and possibly elsewhere). It may have some (positive) impact on production builds too.

> **Note**
> No new code was created in this PR. Functions and other values have been moved and the imports have been updated accordingly.

The CodeMirror query input and the Monaco query input share some code, because they essentially do the same thing.

In most cases where where we import the Monaco editor, we only use it for type declarations, which are removed by Typescript.

But in `client/shared/src/search/query/completion.ts` we actually use it for runtime values, and this file is also used via two different paths by the CodeMirror query input.

This commit separates the code used by the CodeMirror implementation (and also by the Monaco implementation) into separate files, so that the transitive dependency to Monaco is removed.

While a larger refactor/cleanup of the whole query input code is still necessary, this is an easy change to improve loading time and bundle size in development builds where we (presumably?) don't do tree shaking and other optimizations

Size of sg_CodeMirroQuerryInput.chunk.js:
  - before: 19.84 MB
  - after: 2.6 MB

Runtime of some arbitrary integration test that loads a file page:
  - before: 9.4s
  - after: 7.2s

### CODE REVIEW

**Comment by fkling** [comment]
Add @umpox and @vovakulikov as reviewers because bundle size ;)

File: client/shared/src/search/query/completion-utils.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { escapeRegExp } from 'lodash'
```
Comment (novoselrok) [inline client/shared/src/search/query/completion-utils.ts:1]: Optional: maybe it's worth it to add a warning comment to not import Monaco in *-utils files.

**Review by novoselrok — APPROVED** [review]

File: client/shared/src/search/query/completion-utils.ts (hunk 1 above)
Comment (fkling) [reply client/shared/src/search/query/completion-utils.ts:1]: Good idea. I mean, it's OK as long as it's only used for type information. But this whole part needs to cleaned up and properly separate generic search input code from implementation specific (Monaco, CodeMirror) specific code.
I hope I get to that rather sooner than later.

**Review by vovakulikov — APPROVED** [review]
LGTM, Thanks for measuring bundle size-related tracking. About implementation separation I think I would prefer more explicit way based on directory structure instead. For example, 

```
/search/query/
  /code-mirror
  /monaco
  /core
  index.ts
```

But this is my personal opinion; I think the comments at the beginning of the file are also a good enough idea for this kind of separation. Thanks for doing this @fkling

**Comment by fkling** [comment]
>  About implementation separation I think I would prefer more explicit way based on directory structure instead.

Very much agreed and we'll end up there eventually. I want to take some more time and get a better overview over all the code involved before deciding upon a specific structure.
